u_institution

dy2403 opened the pull request:
Closes #423 
Hi professor @sbillinge,
Can you review my initial tests?

sbillinge commented on tests/outputs/u_institutions/institutions.yaml:
```diff
@@ -0,0 +1,38 @@
+"columbiau":
```
this is yaml so I don't think these things should be in quotes.

dy2403 replied:
done!

sbillinge commented on tests/outputs/u_institutions/institutions.yaml:
```diff
@@ -0,0 +1,38 @@
+          - "Fu Foundation School of Engineering " "and Applied Science"
+  "state": "NY"
+  "street": "500 W 120th St"
+  "updated": "2020-05-30"
```
what should happen to this when you update?  I think this may be right because in the tests it makes "today's" date into a fixed date so the tests can always pass, and that date may be this one, but please just check.

dy2403 replied:
I added a '--date' field and passed a date to the update!

sbillinge commented:
I gave comments, but basically the tests look good so you can start on the code.....

dy2403 commented:
Hi @sbillinge!
I changed the tests to require the user to insert the id of the university (instead of using indexes). 
Also, I added a "--new" so the user can use it when adding a new institution. This way, I thought it could make it simpler to use the helper, without indexes. What do you think about this?
Also, do you want the user to be able to add/update departments? I wasn't in the UCs, but I could add it. 
Can you take a look?

sbillinge commented:
> Hi @sbillinge! Can you take a look at my code?
> I changed the tests to require the user to insert the id of the university (instead of using indexes).
> Also, I added a "--new" so the user can use it when adding a new institution. This way, I thought it could make it simpler to use the helper, without indexes. What do you think about this?
> Also, do you want the user to be able to add/update departments? it wasn't in the UCs, but I could add it.

I think the index workflow is better tbh and if all the helpers work the same way it is easier for people I think.  Actually, the best workflow is that if the id exists the helper just does the updates, but if the ID doesn't exist the helper returns the list and then the user can rerun by using up-arrow and adding ``-i 5`` or whatever.  But in agile programming it is better not to tweak workflow too much until you release the code, then you will get lots of feedback on what people need and want!

Yes, we would like it if all aspects of the institution could be updated with this helper.  If it takes a long time though, don't worry too much.  this is not the most crucial of the helpers (especially since we did so much of this owrk already....)

dy2403 commented:
hi @sbillinge!
Can you please review it?

sbillinge commented on regolith/helpers/u_institutionshelper.py:
```diff
@@ -11,21 +11,35 @@
+    subpi.add_argument("-n","--name",
                         help="Full name of the institution")
+    subpi.add_argument("-i","--index",
+                       help="index of the item in the enumerated list chosen to update",
```
remove "chosen"

sbillinge commented on regolith/helpers/u_institutionshelper.py:
```diff
@@ -11,21 +11,35 @@
-    subpi.add_argument("institution_id", help="id of the institution. e.g.:columbiau")
-    subpi.add_argument("--name",
+    subpi.add_argument("institution_id",
+                       help="id of the institution. e.g.:columbiau")
```
rogue colon?

sbillinge commented on regolith/helpers/u_institutionshelper.py:
```diff
@@ -35,15 +35,15 @@ def subparser(subpi):
                        help="e.g.: physics, apam.")
     subpi.add_argument("--dept_name",
-                       help="Department canonical name. e.g.: Department of Physics. "
+                       help="Department canonical name. e.g. Department of Physics. "
```
actually the correct way to do these is:
```
"...canonical name, e.g., Department of Physics."
```

dy2403 replied:
Ok, sorry! I changed it!!

sbillinge commented on regolith/helpers/u_institutionshelper.py:
```diff
@@ -0,0 +1,188 @@
+
+def subparser(subpi):
+    subpi.add_argument("institution_id",
+                       help="id of the institution, e.g., columbiau.")
```
"id of the institution, e.g., columbiau, or fragment of the id, e.g., col"

sbillinge commented on regolith/helpers/u_institutionshelper.py:
```diff
@@ -0,0 +1,188 @@
+                       help="List of all the different names this "
+                            "institution is known by.")
+    subpi.add_argument("--state",
+                       help="The state where the institution is.")
```
state and zip are required if the country is US btw.  Please could you add words to the help and code functionality to ensure this?  Not required if the institution is abroad

sbillinge commented on regolith/helpers/u_institutionshelper.py:
```diff
@@ -0,0 +1,188 @@
+    subpi.add_argument("--state",
+                       help="The state where the institution is.")
+    subpi.add_argument("--zip",
+                       help="zipcode of the institution.")
```
please see above...add words

dy2403 replied:
done! Though not sure if my wording is ok

sbillinge commented on regolith/helpers/u_institutionshelper.py:
```diff
@@ -0,0 +1,188 @@
+                       help="e.g. physics.")
+    subpi.add_argument("--dept_name",
+                       help="Department canonical name, e.g., Department of Physics. "
+                            "Defaults to 'Department of {department id.}'")
```
this is too much magic so please remove.  add "Required if --dept_id supplied and it is a new department"

dy2403 replied:
done!

sbillinge commented on regolith/helpers/u_institutionshelper.py:
```diff
@@ -0,0 +1,188 @@
+                       help="Department aliases, e.g., dept. of physics.")
+    subpi.add_argument("--school",
+                       help="Full canonical name, e.g., School of Engineering and Applied Science.")
+    subpi.add_argument("--school_id",
```
reverse order with previous to keep consistent with department.  Also make school name required if school id is specified.

dy2403 replied:
done!

sbillinge commented:
@dy2403 I am testing this but it doesn't seem to have the right workflow.

1.  I typed ``rh u_milestone`` and it returned the help messages, so that is correct. (there is a problem with the help for the positional argument because it doesn't mention you can put in a fragment if you don't have the full ID)
1. Then I typed ``rh u_milestone mich``.  I was expecting to get a numbered list of entries like umich, michigan state etc., but it just crashed and said "no ID mich"

dy2403 commented:
> @dy2403 I am testing this but it doesn't seem to have the right workflow.
> 
> 1. I typed `rh u_milestone` and it returned the help messages, so that is correct. (there is a problem with the help for the positional argument because it doesn't mention you can put in a fragment if you don't have the full ID)
> 2. Then I typed `rh u_milestone mich`.  I was expecting to get a numbered list of entries like umich, michigan state etc., but it just crashed and said "no ID mich"

Hi, @sbillinge!
Did you type u_milestone? This is the u_institutions, so maybe that is why it crashed. I tried testing it doing 'much' and it behaved properly:
<img width="932" alt="Screen Shot 2020-07-03 at 4 59 45 PM" src="https://user-images.githubusercontent.com/60674550/86495231-d5c86400-bd4e-11ea-90e3-51d061873200.png">

sbillinge commented:
OK, that was my mistake, sorry!

However,
```
rh u_institutions mich
```
returned
```
loading ..\db\institutions.yml...
dumping institutions...
Please rerun the helper specifying '-n list-index' to update item number
'list-index':
1. mich as a new institution.
2. umichigan      University of Michigan.
```
and missed Michigan State University

If you work on this, could you also change it to u_institution and not
u_institutions as it will only updated an institution one at a time.

Thx,

S

On Fri, Jul 3, 2020 at 4:02 PM Daniela Hikari Yano <notifications@github.com>
wrote:

> @dy2403 <https://github.com/dy2403> I am testing this but it doesn't seem
> to have the right workflow.
>
>    1. I typed rh u_milestone and it returned the help messages, so that
>    is correct. (there is a problem with the help for the positional argument
>    because it doesn't mention you can put in a fragment if you don't have the
>    full ID)
>    2. Then I typed rh u_milestone mich. I was expecting to get a numbered
>    list of entries like umich, michigan state etc., but it just crashed and
>    said "no ID mich"
>
> Hi, @sbillinge <https://github.com/sbillinge>!
> Did you type u_milestone? This is the u_institutions, so maybe that is why
> it crashed. I tried testing it doing 'much' and it behaved properly:
> [image: Screen Shot 2020-07-03 at 4 59 45 PM]
> <https://user-images.githubusercontent.com/60674550/86495231-d5c86400-bd4e-11ea-90e3-51d061873200.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/regro/regolith/pull/495#issuecomment-653665092>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAOWUON6ZHD5TENQPJGNCDRZY2MRANCNFSM4OLYUAOQ>
> .
>


-- 
Professor Simon Billinge
Columbia University
